<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">

  <meta name="generator" content="Hugo 0.58.3" />
  <link rel="canonical" href="https://nekrut.github.io/lab_site/rtpct_mut/">

  

  <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
  <link rel="manifest" href="/site.webmanifest">
  <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#000000">
  <meta name="msapplication-TileColor" content="#ffffff">
  <meta name="theme-color" content="#ffffff">

  <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/lab_site/css/paper.css">

  
  
  <link rel="stylesheet" type="text/css" href="https://nekrut.github.io/lab_site/css/custom.css">

  
  
  <title>Effects of RT and PCR on mutation spectra | AN lab</title>
  <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  <script src="https://unpkg.com/feather-icons"></script>
</head>
  <body>
    <div class="container paper">
      <nav class="border split-nav">
  <div class="nav-brand">
    <h3><a href="/">AN lab</a></h3>
  </div>
  <div class="collapsible">
    <input id="collapsible1" type="checkbox" name="collapsible1">
    <button>
    <label for="collapsible1">
        <div class="bar1"></div>
        <div class="bar2"></div>
        <div class="bar3"></div>
      </label>
    </button>
    <div class="collapsible-body">
      <ul class="inline">
      
        <li><a href="http://galaxyproject.org">Galaxy</a></li>
      
        <li><a href="/lab_site/people/">People</a></li>
      
        <li><a href="/lab_site/publications/">Publications</a></li>
      
        <li><a href="/lab_site/post/">Blog</a></li>
      
      </ul>
    </div>
  </div>
</nav>
      <main>
        

<h1 class="post-title">Effects of RT and PCR on mutation spectra</h1>


<strong>Publish date: </strong>Oct 10, 2019
<br>
Author: Anton
<br>

  <strong>Tags: </strong>
  
    <a href="https://nekrut.github.io/lab_site/tags/pcr">PCR</a>
  
    <a href="https://nekrut.github.io/lab_site/tags/rna-viruses">RNA viruses</a>
  
    <a href="https://nekrut.github.io/lab_site/tags/rt">RT</a>
  
    <a href="https://nekrut.github.io/lab_site/tags/mutation">mutation</a>
  



  

<div class="child-borders">
  <div class="toc border" aria-labelledby="toc-heading">
    <h2 id="toc-heading">Table of contents</h2>
    <ol>
      
        <li>
          
          
          
          
          <a href="#design">
            Design
          </a>
        </li>
      
        <li>
          
          
          
          
          <a href="#results">
            Results
          </a>
        </li>
      
    </ol>
  </div>
</div>





<p class="sm col background-warning">
  
This is one of those "i-write-it-for-myself" notes about a paper that I really needed to understand

</p>

<p>This is summary of <a href="http://dx.doi.org/10.1186/s12864-015-1456-x">paper</a> by Orton et al. 2015 &ldquo;Distinguishing low frequency mutations from RT-PCR and sequence errors in viral deep sequencing data&rdquo;</p>

<p>In this manuscript the authors investigated the effects of various stages of sample preparation on the frequency of artifactual mutations. This is particularly interesting because it is based on re-sequencing on a single stranded RNA virus - a procedure that necessarily involves the Reverse Transcription (RT) step. It is therefore applicable to data generated for other (not necessarily single stranded RNA viruses such as HIV-1).</p>

<h2 id="design">Design</h2>

<p>In this paper a cloned partial genome of the Foot and Mouth Disease Virus (FMDV) was sequenced using four strategies:</p>

<div class="card">
  <img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12864-015-1456-x/MediaObjects/12864_2015_1456_Fig2_HTML.gif" width="600px" alt="Fig.2">

  <div class="card-body">
    <p class="card-text"><b>Fig. 2 |</b> Control sample processing and genome targets. (a) is a schematic of the four different control samples and the processing that each one has undergone. The numbers in brackets represent the number of PCR cycles used during PCR steps, the (2) label on reverse transcription denotes that there was a preceding transcription step. (b) is a schematic of the DNA plasmid used, the genomic regions each of the control samples covers, and the regions that are directly comparable between all controls and used for later analyses. From <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1456-x">Orton et al. 2015</a></p>
  </div>
</div>

<p>These strategies were:</p>

<ol>
<li><kbd>Seq</kbd> - sequencing linearized plasmid directly (immediately initiating library prep);</li>
<li><kbd>PCR-low</kbd> - amplifying linearized plasmid using two primer pairs in two 19-cycle PCR reactions and then initiating the library prep;</li>
<li><kbd>PCR-high</kbd> - diluting linearized plasmid and then amplifying it using two primer pairs in two 39-cycle PCR reactions and then initiating the library prep;</li>
<li><kbd>RT + PCR</kbd> - transcribing the linearized plasmid, reverse transcribing it and then amplifying it using two primer pairs in two 39-cycle PCR reactions and then initiating the library prep.</li>
</ol>

<p>The sequencing was done in Illumina GAIIx yielding 73 bp single-end reads (see <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJEB860">PRJEB8601</a>).</p>

<h2 id="results">Results</h2>

<h3 id="mutation-spectra">Mutation spectra</h3>

<p>The mutation spectra obtained from this experiment are show in Fig. 3 of the manuscript. One can see the increase in the number of processing steps shifts mutations towards higher frequency bins:</p>

<p><div class="card">
  <img src="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2Fs12864-015-1456-x/MediaObjects/12864_2015_1456_Fig3_HTML.gif" width="600px" class="no-border alt="Fig.3>
  <div class="card-body">
    <p class="card-text"><b>Fig. 3 |</b> Experimental data mutation spectra. The mutation spectrum of each of the experimental samples: Seq (blue), PCR-Low (red), PCR-High (green) and RT-PCR (purple) along with a real FMDV sample (black) from an infected cow (A5-7DPFC-PB in [18]). Each genome position considered is placed into a discrete bin based on its mismatch (to the reference) frequency. The x-axis represents the mismatch frequency of the bin whilst the y-axis represents the number of genome positions that are in that bin. Both axes are presented on a log10 scale. From <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1456-x">Orton et al. 2015</a></p>
  </div>
</div></p>

<h3 id="variant-frequency-thresholds">Variant frequency thresholds</h3>

<p>This is where it gets interesting:</p>

<pre><code>             25%     50%     75%     95%   100%
-------------------------------------------------
Seq.     0.0119% 0.0183% 0.0268% 0.0466% 0.276%
PCR-Low	 0.0261% 0.0371% 0.0518% 0.0793% 0.4074%
PCR-High 0.0323% 0.0457% 0.0623% 0.1004% 0.3755%
RT-PCR.  0.0368% 0.0526% 0.0793% 0.1403% 0.6571%
Real.    0.0309% 0.0510% 0.0916% 0.2113% 100
-------------------------------------------------
This table contains the 25th, median 50th, 75th, 95th and the maximum
100th percentiles of genome position mismatch frequencies in each of
the four samples.
</code></pre>

<p>The maximum observed mutation frequency for RT-PCR is 0.66% and according to Fig 3 (not Fig. 4 as stated in the text) the real genome starts to deviate from RT-PCR mutation frequency in 0.157-0.324 frequency bin. They further state:</p>

<blockquote>
<p>A total of 226 variants are observed above this threshold in the RT-PCR control sample, whilst 516 variants are observed in the real FMDV sample, 290 of which are therefore likely to be true. A threshold of 0.66% would correctly identify all the errors (100% specificity), but would only identify 32 of the 290 likely variants in the real FMDV sample as true. Lowering the threshold by half to 0.3% results in a doubling of the number of likely true variants identified to 75 at a relatively small cost with specificity only dropping to 99.86%. Whilst at a threshold of 0.15%, specificity drops to 96% with 226 errors falsely identified as true, but all 290 are the likely true variants also identified. It is important to note that a threshold will be dictated by the actual sample processing used and must also be taken in context. The fidelity of the enzymes used and the number of the PCR cycles will highly influence threshold setting, whilst the coverage of a genome position and the quality of itâ€™s aligned reads will influence a threshold in a site specific manner.</p>
</blockquote>

<p>and finally:</p>

<blockquote>
<p>From the data itself we identified a minimum frequency of ~0.5% above which one can be reasonably confident that an observed mutation is real.</p>
</blockquote>

  

      </main>
      
  
  <script>
  window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
  ga('create', 'UA-123456789-1', 'auto');
  ga('send', 'pageview');
  </script>
  <script async src='https://www.google-analytics.com/analytics.js'></script>
  

  </div>
  </body>
</html>