<!doctype html>

<html lang="en" class="h-100">
  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  <meta name="generator" content="Hugo 0.58.3" />
  <link rel="stylesheet" href="https://nekrut.github.io/lab_site/css/bootstrap.min.css">
  
  
  <title>History of pBR322 | AN LAB</title>
  <style>
.container {
  max-width: 700px;
}
#nav a {
  font-weight: bold;
  color: inherit;
}
#nav a.nav-link-active {
  background-color: #212529;
  color: #fff;
}
#nav-border {
  border-bottom: 1px solid #212529;
}
#main {
  margin-top: 1em;
  margin-bottom: 4em;
}
#home-jumbotron {
  background-color: inherit;
}
#footer .container {
  padding: 1em 0;
}
#footer a {
  color: inherit;
  text-decoration: underline;
}
.font-125 {
  font-size: 125%;
}
.tag-btn {
  margin-bottom: 0.3em;
}
pre {
  background-color: #f5f5f5;
  border: 1px solid #ccc;
  border-radius: 4px;
  padding: 16px;
}
pre code {
  padding: 0;
  font-size: inherit;
  color: inherit; 
  background-color: transparent;
  border-radius: 0;
}
code {
  padding: 2px 4px;
  font-size: 90%;
  color: #c7254e;
  background-color: #f9f2f4;
  border-radius: 4px;
}
img,
iframe,
embed,
video,
audio {
  max-width: 100%;
}
.card-img,
.card-img-top,
.card-img-bottom {
  width: initial;
}
</style>
</head>
  <body class="d-flex flex-column h-100">
    <div id="nav-border" class="container">
  <nav id="nav" class="nav justify-content-center">
  
  
  
    
    
      
      
      
      
        
      
    
    
    <a class="nav-link " href="/lab_site/"><i data-feather="home"></i> Home</a>
  
    
    
      
      
      
      
        
      
    
    
    <a class="nav-link " href="/lab_site/people/"><i data-feather="smile"></i> People</a>
  
    
    
      
      
      
      
        
      
    
    
    <a class="nav-link " href="/lab_site/pubs/"><i data-feather="book-open"></i> Pubs</a>
  
    
    
      
      
      
      
       
        
        
      
    
    
    <a class="nav-link nav-link-active" href="/lab_site/post/"><i data-feather="file-text"></i> Blog</a>
  
    
    
      
      
      
      
        
      
    
    
    <a class="nav-link " href="/lab_site/coord/"><i data-feather="map-pin"></i> Coord</a>
  
    
    
      
      
      
      
        
      
    
    
    <a class="nav-link " href="http://galaxyproject.org"><i data-feather="arrow-up"></i> Galaxy</a>
  
  </nav>
</div>
    <div class="container">
      <main id="main">
        

<h1>History of pBR322</h1>


<i data-feather="calendar"></i> <time datetime="2019-08-23">Aug 23, 2019</time>

  <br>
  <i data-feather="tag"></i>
  
  
  <a class="btn btn-sm btn-outline-dark tag-btn" href="https://nekrut.github.io/lab_site/tags/plasmid">plasmid</a>
  
  
  <a class="btn btn-sm btn-outline-dark tag-btn" href="https://nekrut.github.io/lab_site/tags/pbr322">pBR322</a>
  
  
  <a class="btn btn-sm btn-outline-dark tag-btn" href="https://nekrut.github.io/lab_site/tags/history">history</a>
  

<br><br>


<p>by AN</p>

<hr />

<div class="alert alert-warning" role="alert">
We take cloning vectors for given but most people do not know their history. The author is one of these people. Yet because we use pBR322 extensively, I wanted to know its origins. This is a condensed version of that investigation. 
</div>

<p>Stanley Cohen and co-workers (<a href="http://dx.doi.org/10.1073/pnas.70.11.3240">Cohen et al. 1973</a>) constructed first functional plasmids by transforming <em>E. coli</em> with fragments produced by treating R6-5 and pSC101 plasmids using Eco RI restriction enzyme. They effectively showed that functional extrachromosomal replicons can be obtained by joining plasmid DNA of completely different origins. R6-5 is a variant of R6 plasmid that has lost Tetracycline (Tc) resistance marker (<a href="https://www.ncbi.nlm.nih.gov/pubmed/4580676">Sharp et al. 1973</a>). R6 itself is an F-factor, a sex factor-like plasmid. pSC101 (SC stands for Stanley Cohen) was initially thought to be derived from R6-5 by controlled shearing of R6-5 DNA. This notion was later disputed and contamination with some other plasmid DNA (such as SP219, a <em>Salmonella panama plasmid</em>) was later suggested as the potential origin of pSC101 (<a href="http://jb.asm.org/cgi/pmidlookup?view=long&amp;pmid=334752">Cohen and Chang 1977</a>).</p>

<p>Colicinogenic factor E1 (ColE1) was isolated from <em>E. coli</em>  strain JC411 as a molecule sedimented as a 23 S particle. Compared to the first cloning vector, pSC101, ColE1 plasmid offered several benefits as a potential cloning vector: relatively high copy number of ~24 estimated based on the ratio of plasmid-to-chriomosomal DNA (<a href="https://jb.asm.org/content/jb/110/3/1135.full.pdf">Clewell and Helinski 1972</a>), small size, and the presence of a single EcoRI restriction site. Hershfield (<a href="http://dx.doi.org/10.1073/pnas.71.9.3455">Hershfield et al. 1974</a>) constructed initial ColE1-based vectors by incorporating Trp operon and Kanamycin resistance genes to create pVH and pML2 recombinant plasmid families (using pJC411 ColE1 plasmid; see (<a href="http://dx.doi.org/10.1016/0022-2836(68)90374-4">Bazaral and Helinski 1968</a>)). One of the recombinants, pVH51, was analyzed in great detail (<a href="http://jb.asm.org/cgi/pmidlookup?view=long&amp;pmid=770430">Hershfield et al. 1976</a>). It exhibited a higher copy number compared to the original ColE1, did not produce colicin E1 but conferred immunity against it. Heteroduplex analysis showed that ~90% of pVH51 hybridized with ~50% of ColE1. Bolivar et al. (<a href="https://linkinghub.elsevier.com/retrieve/pii/0378-1119(77)90074-9">Bolivar et al. 1977</a>) set out to improve upon the ColE1-derived recombinant plasmid. They used a previously isolated pVH51-like plasmid pMB8 (<a href="https://europepmc.org/abstract/med/776705">Betlach et al. 1976</a>) to construct pMB9 containing tetracycline resistance marker (Tcr). It was achieved by ligating EcoRI* digest products of pSC101 (mentioned above) and EcoRI digest products of pMB8 (note the distinction between EcoRI and EcoRI*, where * indicates so-called &ldquo;star activity&rdquo; or the ability of EcoRI to recognize a <code>NAATTN</code> site instead of the typical <code>GAATTC</code> in conditions of lowered ionic strengths or altered pH (<a href="https://onlinelibrary.wiley.com/doi/pdf/10.1016/0014-5793(78)80400-1">Mayer 1978</a>)). pMB9 was a substantial improvement as it allowed cloning of foreign DNA by inserting it at a single EcoRI and selecting recombinants based on tetracycline resistance. However cloning at HindIII, BamHI, and SalI sites, a set of restriction enzymes having a single recognition site within pMB9, disrupted tetracycline resistance gene and as a result screening for recombinants was only possible by screening for colicin E1 immunity &ndash; a tedious  and unreliable procedure. To augment this limitation pMB9 was co-transformed into <em>E. coli</em> cells (strain RRI) together with pRSF2124-a plasmid containing ampicillin resistance marker TnA. The translocation of TnA locus from pRS2124 (<a href="https://www.ncbi.nlm.nih.gov/pubmed/1045010">So et al. 1975</a>) onto pMB9 gave rise to multiple Tet and Amp resistant clones two of which, pBR312 and pBR26 were further characterized. While these two plasmids contained the second selection marker they were large and contained two BamHI sites limiting their utility. Bolivar and Rodriguez attempted to remove the second BamHI site and reduce the size of pBR312 by partial EcoRI* digestion followed by ligation and transformation. They isolated three smallest clones pBR313, pBR315, and pBR316. The utility of one of these plasmids, pBR313, was highlighted in great detail.</p>

<p>Subsequently Bolivar and Rodriguez created (<kbd>B</kbd> and <kbd>R</kbd> of <kbd>pBR322</kbd>) pBR322, a streamlined version of pBR313 (<a href="https://linkinghub.elsevier.com/retrieve/pii/0378-1119(77)90000-2">Bolivar et al. 1977</a>). pBR322 was significantly (&gt;50%) smaller than pBR322 and contained a series of additional restriction sites. pBR322 was created in several steps. First, pBR313 was digested with PstI, fragments were ligated, transformed into <em>E. coli</em> RRI and TcR transformants were selected. This procedure yielded pBR318 plasmid. A similar procedure were bacterial cells were transformed with unligated EcoRII fragments of pBR313 and screened for ApR colonies yielded pBR320. These two plasmids, pBR318 and pBR320 were then digested with restriction enzymes, mixed, ligated and transformed into cells. This gave rise to numerous Tc<sup>R</sup>Ap<sup>R</sup>Col<sup>S</sup> clones including those carrying plasmids pBR321 and pBR322. pBR322 was the smallest and was chosen for subsequent analysis.  As these authors note in the discussion section of their manuscript <em>“A new amplifiable cloning vehicle, the plasmid pBR322, which improves the cloning characteristics of the parental plasmid pBR313, has been constructed by in vitro recombination techniques.”</em></p>

<hr />

<p><img src="/lab_site/images/pbr322.png" alt="" /></p>

<p><small>Graphical history of pBR322 by <a href="http://dx.doi.org/10.1101/SQB.1979.043.01.013">Sutcliffe 1979</a></small></p>

<hr />

<h2 id="references">References</h2>

<ol>
<li>Sutcliffe JG. Complete nucleotide sequence of the Escherichia coli plasmid pBR322. Cold Spring Harb Symp Quant Biol. 1979;43 Pt 1:77–90. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=383387">383387</a></li>
<li>Cohen SN, Chang AC, Boyer HW, Helling RB. Construction of biologically functional bacterial plasmids in vitro. Proc Natl Acad Sci U S A. 1973 Nov;70(11):3240–3244. PMCID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=PMC427208">PMC427208</a></li>
<li>Sharp PA, Cohen SN, Davidson N. Electron microscope heteroduplex studies of sequence relations among plasmids of Escherichia coli. II. Structure of drug resistance &reg; factors and F factors. J Mol Biol. Elsevier; 1973 Apr 5;75(2):235–255. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=4580676">4580676</a></li>
<li>Cohen SN, Chang AC. Revised interpretation of the origin of the pSC101 plasmid. J Bacteriol. 1977 Nov;132(2):734–737. PMCID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=PMC221918">PMC221918</a></li>
<li>Clewell DB, Helinski DR. Effect of growth conditions on the formation of the relaxation complex of supercoiled ColE1 deoxyribonucleic acid and protein in Escherichia coli. J Bacteriol. Am Soc Microbiol; 1972 Jun;110(3):1135–1146. PMCID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=PMC247537">PMC247537</a></li>
<li>Hershfield V, Boyer HW, Yanofsky C, Lovett MA, Helinski DR. Plasmid ColEl as a molecular vehicle for cloning and amplification of DNA. Proc Natl Acad Sci U S A. 1974 Sep;71(9):3455–3459. PMCID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=PMC433792">PMC433792</a></li>
<li>Bazaral M, Helinski DR. Circular DNA forms of colicinogenic factors E1, E2 and E3 from Escherichia coli. J Mol Biol. 1968 Sep 14;36(2):185–194. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=4939624">4939624</a></li>
<li>Hershfield V, Boyer HW, Chow L, Helinski DR. Characterization of a mini-ColC1 plasmid. J Bacteriol. 1976 Apr;126(1):447–453. PMCID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=PMC233301">PMC233301</a></li>
<li>Bolivar F, Rodriguez RL, Betlach MC, Boyer HW. Construction and characterization of new cloning vehicles. I. Ampicillin-resistant derivatives of the plasmid pMB9. Gene. 1977;2(2):75–93. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=344136">344136</a></li>
<li>Betlach M, Hershfield V, Chow L, Brown W, Goodman H, Boyer HW. A restriction endonuclease analysis of the bacterial plasmid controlling the ecoRI restriction and modification of DNA. Fed Proc. 1976 Jul;35(9):2037–2043. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=776705">776705</a></li>
<li>Mayer H. Optimization of the EcoRI*-activity of EcoRI endonuclease. FEBS Lett. Wiley Online Library; 1978;90(2):341–344.</li>
<li>So M, Gill R, Falkow S. The generation of a ColE1-Apr cloning vehicle which allows detection of inserted DNA. Mol Gen Genet. 1975 Dec 30;142(3):239–249. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=1045010">1045010</a></li>
<li>Bolivar F, Rodriguez RL, Greene PJ, Betlach MC, Heyneker HL, Boyer HW, Crosa JH, Falkow S. Construction and characterization of new cloning vehicles. II. A multipurpose cloning system. Gene. 1977;2(2):95–113. PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=344137">344137</a></li>
</ol>



      </main>
    </div>
    
<footer id="footer" class="mt-auto text-center text-muted">
  <div class="container">
    AN LAB | CC-BY
  </div>
</footer>

    <script src="https://nekrut.github.io/lab_site/js/feather.min.js"></script>
<script>
  feather.replace()
</script>


<script src="https://nekrut.github.io/lab_site/js/jquery-3.3.1.slim.min.js"></script>
<script src="https://nekrut.github.io/lab_site/js/bootstrap.bundle.min.js"></script>

    
  
  <script>
  window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
  ga('create', 'UA-123456789-1', 'auto');
  ga('send', 'pageview');
  </script>
  <script async src='https://www.google-analytics.com/analytics.js'></script>
  

  </body>
</html>