---
date: "2019-10-11"
title: "Publications"
---

In ascending order from 1998 to just about <a href="#now">now</a>:

1. **Cytosolic isocitrate dehydrogenase in humans, mice, and voles and phylogenetic analysis of the enzyme family.**  Nekrutenko A, Hillis DM, Patton JC, Bradley RD, Baker RJ.   [*Mol Biol Evol.  1998*](http://www.ncbi.nlm.nih.gov/pubmed/9866202)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='9866202'></div> 
1. **Representational difference analysis to distinguish cryptic species.**  Nekrutenko A, Makova KD, Chesser RK, Baker RJ.   [*Mol Ecol.  1999*](http://www.ncbi.nlm.nih.gov/pubmed/10447864)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='10447864'></div> 
1. **Isolation of binary species-specific PCR-based markers and their value for diagnostic applications.**  Nekrutenko A, Makova KD, Baker RJ.   [*Gene.  2000*](http://www.ncbi.nlm.nih.gov/pubmed/10831837)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='10831837'></div> 
1. **Evolution of microsatellite alleles in four species of mice (genus Apodemus).**  Makova KD, Nekrutenko A, Baker RJ.   [*J Mol Evol.  2000*](http://www.ncbi.nlm.nih.gov/pubmed/10948273)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='10948273'></div> 
1. **Assessment of compositional heterogeneity within and between eukaryotic genomes.**  Nekrutenko A, Li WH.   [*Genome Res.  2000*](http://www.ncbi.nlm.nih.gov/pubmed/11116093)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='11116093'></div> 
1. **Densities, length proportions, and other distributional features of repetitive sequences in the human genome estimated from 430 megabases of genomic sequence.**  Gu Z, Wang H, Nekrutenko A, Li WH.   [*Gene.  2000*](http://www.ncbi.nlm.nih.gov/pubmed/11163965)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='11163965'></div> 
1. **Bushbaby growth hormone is much more similar to nonprimate growth hormones than to rhesus monkey and human growth hormones.**  Adkins RM, Nekrutenko A, Li WH.   [*Mol Biol Evol.  2001*](http://www.ncbi.nlm.nih.gov/pubmed/11141192)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='11141192'></div> 
1. **Evolutionary analyses of the human genome.**  Li WH, Gu Z, Wang H, Nekrutenko A.   [*Nature.  2001*](http://www.ncbi.nlm.nih.gov/pubmed/11237007)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='11237007'></div> 
1.  **Transposable elements are found in a large number of human protein-coding genes.**  Nekrutenko A, Li WH.   [*Trends Genet.  2001*](http://www.ncbi.nlm.nih.gov/pubmed/11672845)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='11672845'></div> 
1.  **The K(A)/K(S) ratio test for assessing the protein-coding potential of genomic regions: an empirical and simulation study.**  Nekrutenko A, Makova KD, Li WH.   [*Genome Res.  2002*](http://www.ncbi.nlm.nih.gov/pubmed/11779845)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='11779845'></div> 
1.  **Signatures of domain shuffling in the human genome.**  Kaessmann H, ZÃ¶llner S, Nekrutenko A, Li WH.   [*Genome Res.  2002*](http://www.ncbi.nlm.nih.gov/pubmed/12421750)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='12421750'></div> 
1.  **Detection of gene duplications and block duplications in eukaryotic genomes.**  Li WH, Gu Z, Cavalcanti AR, Nekrutenko A.   [*J Struct Funct Genomics.  2003*](http://www.ncbi.nlm.nih.gov/pubmed/12836682)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='12836682'></div> 
1.  **Subgenome-specific markers in allopolyploid cotton Gossypium hirsutum: implications for evolutionary analysis of polyploids.**  Nekrutenko A, Baker RJ.   [*Gene.  2003*](http://www.ncbi.nlm.nih.gov/pubmed/12657471)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='12657471'></div> 
1.  **An evolutionary approach reveals a high protein-coding capacity of the human genome.**  Nekrutenko A, Chung WY, Li WH.   [*Trends Genet.  2003*](http://www.ncbi.nlm.nih.gov/pubmed/12801721)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='12801721'></div> 
1.  **Evolutionary dynamics of oncogenes and tumor suppressor genes: higher intensities of purifying selection than other genes.**  Thomas MA, Weston B, Joseph M, Wu W, Nekrutenko A, Tonellato PJ.   [*Mol Biol Evol.  2003*](http://www.ncbi.nlm.nih.gov/pubmed/12716985)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='12716985'></div> 
1.  **ETOPE: Evolutionary test of predicted exons.**  Nekrutenko A, Chung WY, Li WH.   [*Nucleic Acids Res.  2003*](http://www.ncbi.nlm.nih.gov/pubmed/12824366)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='12824366'></div> 
1.  **Comparative genomics.**  Miller W, Makova KD, Nekrutenko A, Hardison RC.   [*Annu Rev Genomics Hum Genet.  2004*](http://www.ncbi.nlm.nih.gov/pubmed/15485342)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='15485342'></div> 
1.  **Genome sequence of the Brown Norway rat yields insights into mammalian evolution.**  Gibbs RA, Weinstock GM, Metzker ML, Muzny DM, Sodergren EJ, Scherer S, Scott G, Steffen D, Worley KC, Burch PE, Okwuonu G, Hines S, Lewis L, DeRamo C, Delgado O, Dugan-Rocha S, Miner G, Morgan M, Hawes A, Gill R, Celera, Holt RA, et al.   [*Nature.  2004*](http://www.ncbi.nlm.nih.gov/pubmed/15057822)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='15057822'></div> 
1.  **Reconciling the numbers: ESTs versus protein-coding genes.**  Nekrutenko A.   [*Mol Biol Evol.  2004*](http://www.ncbi.nlm.nih.gov/pubmed/15034132)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='15034132'></div> 
1.  **Identification of novel exons from rat-mouse comparisons.**  Nekrutenko A.   [*J Mol Evol.  2004*](http://www.ncbi.nlm.nih.gov/pubmed/15693626)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='15693626'></div> 
1.  **Oscillating evolution of a mammalian locus with overlapping reading frames: an XLalphas/ALEX relay.**  Nekrutenko A, Wadhawan S, Goetting-Minesky P, Makova KD.   [*PLoS Genet.  2005*](http://www.ncbi.nlm.nih.gov/pubmed/16110341)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='16110341'></div> 
1.  **Galaxy: a platform for interactive large-scale genome analysis.**  Giardine B, Riemer C, Hardison RC, Burhans R, Elnitski L, Shah P, Zhang Y, Blankenberg D, Albert I, Taylor J, Miller W, Kent WJ, Nekrutenko A.   [*Genome Res.  2005*](http://www.ncbi.nlm.nih.gov/pubmed/16169926)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='16169926'></div> 
1.  **Rapid and asymmetric divergence of duplicate genes in the human gene coexpression network.**  Chung WY, Albert R, Albert I, Nekrutenko A, Makova KD.   [*BMC Bioinformatics.  2006*](http://www.ncbi.nlm.nih.gov/pubmed/16441884)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='16441884'></div> 
1.  **mNSC1 shows no evidence of protein-coding capacity.**  Wilson C, Goetting-Minesky P, Nekrutenko A.   [*Gene.  2006*](http://www.ncbi.nlm.nih.gov/pubmed/16476528)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='16476528'></div> 
1.  **Functionality of unspliced XBP1 is required to explain evolution of overlapping reading frames.**  Nekrutenko A, He J.   [*Trends Genet.  2006*](http://www.ncbi.nlm.nih.gov/pubmed/17034899)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='17034899'></div> 
1.  **A first look at ARFome: dual-coding genes in mammalian genomes.**  Chung WY, Wadhawan S, Szklarczyk R, Pond SK, Nekrutenko A.   [*PLoS Comput Biol.  2007*](http://www.ncbi.nlm.nih.gov/pubmed/17511511)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='17511511'></div> 
1.  **A framework for collaborative analysis of ENCODE data: making large-scale analyses biologist-friendly.**  Blankenberg D, Taylor J, Schenck I, He J, Zhang Y, Ghent M, Veeraraghavan N, Albert I, Miller W, Makova KD, Hardison RC, Nekrutenko A.   [*Genome Res.  2007*](http://www.ncbi.nlm.nih.gov/pubmed/17568012)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='17568012'></div> 
1.  **Rapid asymmetric evolution of a dual-coding tumor suppressor INK4a/ARF locus contradicts its function.**  Szklarczyk R, Heringa J, Pond SK, Nekrutenko A.   [*Proc Natl Acad Sci U S A.  2007*](http://www.ncbi.nlm.nih.gov/pubmed/17652172)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='17652172'></div> 
1.  **Using galaxy to perform large-scale interactive data analyses.**  Taylor J, Schenck I, Blankenberg D, Nekrutenko A.   [*Curr Protoc Bioinformatics.  2007*](http://www.ncbi.nlm.nih.gov/pubmed/18428782)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='18428782'></div> 
1.  **28-way vertebrate alignment and conservation track in the UCSC Genome Browser.**  Miller W, Rosenbloom K, Hardison RC, Hou M, Taylor J, Raney B, Burhans R, King DC, Baertsch R, Blankenberg D, Kosakovsky Pond SL, Nekrutenko A, Giardine B, Harris RS, Tyekucheva S, Diekhans M, Pringle TH, Murphy WJ, Lesk A, Weinstock GM, Lindblad-Toh K, Gibbs RA, et al.   [*Genome Res.  2007*](http://www.ncbi.nlm.nih.gov/pubmed/17984227)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='17984227'></div> 
1.  **Toward the commoditization of translational genomic research: Design and implementation features of the Galaxy genomic workbench.**  Lazarus R, Taylor J, Qiu W, Nekrutenko A.   [*Summit Transl Bioinform.  2008*](http://www.ncbi.nlm.nih.gov/pubmed/21347127)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='21347127'></div> 
1.  **Wheels within wheels: clues to the evolution of the Gnas and Gnal loci.**  Wadhawan S, Dickins B, Nekrutenko A.   [*Mol Biol Evol.  2008*](http://www.ncbi.nlm.nih.gov/pubmed/18842685)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='18842685'></div> 
1.  **Transcriptome of embryonic and neonatal mouse cortex by high-throughput RNA sequencing.**  Han X, Wu X, Chung WY, Li T, Nekrutenko A, Altman NS, Chen G, Ma H.   [*Proc Natl Acad Sci U S A.  2009*](http://www.ncbi.nlm.nih.gov/pubmed/19617558)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='19617558'></div> 
1.  **High-resolution mapping of evolutionary trajectories in a phage.**  Dickins B, Nekrutenko A.   [*Genome Biol Evol.  2009*](http://www.ncbi.nlm.nih.gov/pubmed/20333199)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='20333199'></div> 
1.  **Windshield splatter analysis with the Galaxy metagenomic pipeline.**  Kosakovsky Pond S, Wadhawan S, Chiaromonte F, Ananda G, Chung WY, Taylor J, Nekrutenko A; Galaxy Team..   [*Genome Res.  2009*](http://www.ncbi.nlm.nih.gov/pubmed/19819906)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='19819906'></div> 
1.  **Web-based analysis of (Epi-) genome data using EpiGRAPH and Galaxy.**  Bock C, Von Kuster G, Halachev K, Taylor J, Nekrutenko A, Lengauer T.   [*Methods Mol Biol.  2010*](http://www.ncbi.nlm.nih.gov/pubmed/20238087)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='20238087'></div> 
1.  **Galaxy: a comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences.**  Goecks J, Nekrutenko A, Taylor J; Galaxy Team..   [*Genome Biol.  2010*](http://www.ncbi.nlm.nih.gov/pubmed/20738864)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='20738864'></div> 
1.  **Galaxy: a web-based genome analysis tool for experimentalists.**  Blankenberg D, Von Kuster G, Coraor N, Ananda G, Lazarus R, Mangan M, Nekrutenko A, Taylor J.   [*Curr Protoc Mol Biol.  2010*](http://www.ncbi.nlm.nih.gov/pubmed/20069535)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='20069535'></div> 
1.  **Complete Khoisan and Bantu genomes from southern Africa.**  Schuster SC, Miller W, Ratan A, Tomsho LP, Giardine B, Kasson LR, Harris RS, Petersen DC, Zhao F, Qi J, Alkan C, Kidd JM, Sun Y, Drautz DI, Bouffard P, Muzny DM, Reid JG, Nazareth LV, Wang Q, Burhans R, Riemer C, Wittekindt NE, et al.   [*Nature.  2010*](http://www.ncbi.nlm.nih.gov/pubmed/20164927)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='20164927'></div> 
1.  **Manipulation of FASTQ data with Galaxy.**  Blankenberg D, Gordon A, Von Kuster G, Coraor N, Taylor J, Nekrutenko A; Galaxy Team..   [*Bioinformatics.  2010*](http://www.ncbi.nlm.nih.gov/pubmed/20562416)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='20562416'></div> 
1.  **Galaxy CloudMan: delivering cloud compute clusters.**  Afgan E, Baker D, Coraor N, Chapman B, Nekrutenko A, Taylor J.   [*BMC Bioinformatics.  2010*](http://www.ncbi.nlm.nih.gov/pubmed/21210983)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='21210983'></div> 
1.  **A massively parallel sequencing approach uncovers ancient origins and high genetic variability of endangered Przewalski's horses.**  Goto H, Ryder OA, Fisher AR, Schultz B, Kosakovsky Pond SL, Nekrutenko A, Makova KD.   [*Genome Biol Evol.  2011*](http://www.ncbi.nlm.nih.gov/pubmed/21803766)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='21803766'></div> 
1.  **Dynamics of mitochondrial heteroplasmy in three families investigated via a repeatable re-sequencing study.**  Goto H, Dickins B, Afgan E, Paul IM, Taylor J, Makova KD, Nekrutenko A.   [*Genome Biol.  2011*](http://www.ncbi.nlm.nih.gov/pubmed/21699709)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='21699709'></div> 
1.  **Integrating diverse databases into an unified analysis framework: a Galaxy approach.**  Blankenberg D, Coraor N, Von Kuster G, Taylor J, Nekrutenko A; Galaxy Team..   [*Database (Oxford).  2011*](http://www.ncbi.nlm.nih.gov/pubmed/21531983)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='21531983'></div> 
1.  **Making whole genome multiple alignments usable for biologists.**  Blankenberg D, Taylor J, Nekrutenko A; Galaxy Team..   [*Bioinformatics.  2011*](http://www.ncbi.nlm.nih.gov/pubmed/21775304)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='21775304'></div> 
1.  **Harnessing cloud computing with Galaxy Cloud.**  Afgan E, Baker D, Coraor N, Goto H, Paul IM, Makova KD, Nekrutenko A, Taylor J.   [*Nat Biotechnol.  2011*](http://www.ncbi.nlm.nih.gov/pubmed/22068528)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='22068528'></div> 
1.  **Using Galaxy to perform large-scale interactive data analyses.**  Hillman-Jackson J, Clements D, Blankenberg D, Taylor J, Nekrutenko A; Galaxy Team..   [*Curr Protoc Bioinformatics.  2012*](http://www.ncbi.nlm.nih.gov/pubmed/22700312)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='22700312'></div> 
1.  **Next-generation sequencing data interpretation: enhancing reproducibility and accessibility.**  Nekrutenko A, Taylor J.   [*Nat Rev Genet.  2012*](http://www.ncbi.nlm.nih.gov/pubmed/22898652)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='22898652'></div> 
1.  **NGS analyses by visualization with Trackster.**  Goecks J, Coraor N; Galaxy Team., Nekrutenko A, Taylor J.   [*Nat Biotechnol.  2012*](http://www.ncbi.nlm.nih.gov/pubmed/23138293)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='23138293'></div> 
1.  **Web-based visual analysis for high-throughput genomics.**  Goecks J, Eberhard C, Too T; Galaxy Team., Nekrutenko A, Taylor J.   [*BMC Genomics.  2013*](http://www.ncbi.nlm.nih.gov/pubmed/23758618)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='23758618'></div> 
1.  **Ten simple rules for reproducible computational research.**  Sandve GK, Nekrutenko A, Taylor J, Hovig E.   [*PLoS Comput Biol.  2013*](http://www.ncbi.nlm.nih.gov/pubmed/24204232)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='24204232'></div> 
1.  **The anatomy of successful computational biology software.**  Altschul S, Demchak B, Durbin R, Gentleman R, Krzywinski M, Li H, Nekrutenko A, Robinson J, Rasband W, Taylor J, Trapnell C.   [*Nat Biotechnol.  2013*](http://www.ncbi.nlm.nih.gov/pubmed/24104757)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='24104757'></div> 
1.  **RNA-DNA differences in human mitochondria restore ancestral form of 16S ribosomal RNA.**  Bar-Yaacov D, Avital G, Levin L, Richards AL, Hachen N, Rebolledo Jaramillo B, Nekrutenko A, Zarivach R, Mishmar D.   [*Genome Res.  2013*](http://www.ncbi.nlm.nih.gov/pubmed/23913925)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='23913925'></div> 
1.  **Dissemination of scientific software with Galaxy ToolShed.**  Blankenberg D, Von Kuster G, Bouvier E, Baker D, Afgan E, Stoler N; Galaxy Team., Taylor J, Nekrutenko A.   [*Genome Biol.  2014*](http://www.ncbi.nlm.nih.gov/pubmed/25001293)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='25001293'></div> 
1.  **Controlling for contamination in re-sequencing studies with a reproducible web-based phylogenetic approach.**  Dickins B, Rebolledo-Jaramillo B, Su MS, Paul IM, Blankenberg D, Stoler N, Makova KD, Nekrutenko A.   [*Biotechniques.  2014*](http://www.ncbi.nlm.nih.gov/pubmed/24641477)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='24641477'></div> 
1.  **Wrangling Galaxy's reference data.**  Blankenberg D, Johnson JE; Galaxy Team., Taylor J, Nekrutenko A.   [*Bioinformatics.  2014*](http://www.ncbi.nlm.nih.gov/pubmed/24585771)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='24585771'></div> 
1.  **Maternal age effect and severe germ-line bottleneck in the inheritance of human mitochondrial DNA.**  Rebolledo-Jaramillo B, Su MS, Stoler N, McElhoe JA, Dickins B, Blankenberg D, Korneliussen TS, Chiaromonte F, Nielsen R, Holland MM, Paul IM, Nekrutenko A, Makova KD.   [*Proc Natl Acad Sci U S A.  2014*](http://www.ncbi.nlm.nih.gov/pubmed/25313049)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='25313049'></div> 
1.  **Online resources for genomic analysis using high-throughput sequencing.**  Blankenberg D, Taylor J, Nekrutenko A.   [*Cold Spring Harb Protoc.  2015*](http://www.ncbi.nlm.nih.gov/pubmed/25655493)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='25655493'></div> 
1.  **StructureFold: genome-wide RNA secondary structure mapping and reconstruction in vivo.**  Tang Y, Bouvier E, Kwok CK, Ding Y, Nekrutenko A, Bevilacqua PC, Assmann SM.   [*Bioinformatics.  2015*](http://www.ncbi.nlm.nih.gov/pubmed/25886980)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='25886980'></div> 
1.  **Integrative genomic analysis by interoperation of bioinformatics tools in GenomeSpace.**  Qu K, Garamszegi S, Wu F, Thorvaldsdottir H, Liefeld T, Ocana M, Borges-Rivera D, Pochet N, Robinson JT, Demchak B, Hull T, Ben-Artzi G, Blankenberg D, Barber GP, Lee BT, Kuhn RM, Nekrutenko A, Segal E, Ideker T, Reich M, Regev A, Chang HY, et al.   [*Nat Methods.  2016*](http://www.ncbi.nlm.nih.gov/pubmed/26780094)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='26780094'></div> 
1.  **The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2016 update.**  Afgan E, Baker D, van den Beek M, Blankenberg D, Bouvier D, Äech M, Chilton J, Clements D, Coraor N, Eberhard C, GrÃ¼ning B, Guerler A, Hillman-Jackson J, Von Kuster G, Rasche E, Soranzo N, Turaga N, Taylor J, Nekrutenko A, Goecks J.   [*Nucleic Acids Res.  2016*](http://www.ncbi.nlm.nih.gov/pubmed/27137889)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='27137889'></div> 
1.  **Streamlined analysis of duplex sequencing data with Du Novo.**  Stoler N, Arbeithuber B, Guiblet W, Makova KD, Nekrutenko A.   [*Genome Biol.  2016*](http://www.ncbi.nlm.nih.gov/pubmed/27566673)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='27566673'></div> 
1.  **Jupyter and Galaxy: Easing entry barriers into complex data analyses for biomedical researchers.**  GrÃ¼ning BA, Rasche E, Rebolledo-Jaramillo B, Eberhard C, Houwaart T, Chilton J, Coraor N, Backofen R, Taylor J, Nekrutenko A.   [*PLoS Comput Biol.  2017*](http://www.ncbi.nlm.nih.gov/pubmed/28542180)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='28542180'></div> 
1.  **Biology Needs Evolutionary Software Tools: Let's Build Them Right.**  Nekrutenko A, Team G, Goecks J, Taylor J, Blankenberg D.   [*Mol Biol Evol.  2018*](http://www.ncbi.nlm.nih.gov/pubmed/29688462)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='29688462'></div> 
1.  **Practical Computational Reproducibility in the Life Sciences.**  GrÃ¼ning B, Chilton J, KÃ¶ster J, Dale R, Soranzo N, van den Beek M, Goecks J, Backofen R, Nekrutenko A, Taylor J.   [*Cell Syst.  2018*](http://www.ncbi.nlm.nih.gov/pubmed/29953862)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='29953862'></div> 
1.  **Community-Driven Data Analysis Training for Biology.**  Batut B, Hiltemann S, Bagnacani A, Baker D, Bhardwaj V, Blank C, Bretaudeau A, Brillet-GuÃ©guen L, Äech M, Chilton J, Clements D, Doppelt-Azeroual O, Erxleben A, Freeberg MA, Gladman S, Hoogstrate Y, Hotz HR, Houwaart T, Jagtap P, LariviÃ¨re D, Le CorguillÃ© G, Manke T, et al.   [*Cell Syst.  2018*](http://www.ncbi.nlm.nih.gov/pubmed/29953864)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='29953864'></div> 
1.  **Family reunion via error correction: An efficient analysis of duplex sequencing data**  Stoler N, Arbeithuber B, Povysil M, Heinzl M, Salazar R, Makova K, Tiemann-Boege I, Nekrutenko A   [*bioRxiv 2018*](https://www.biorxiv.org/content/10.1101/469106v1)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-doi='10.1101/469106'></div> 
1.  **Understanding trivial challenges of microbial genomics: An assembly example**  DeLariviere D, Mei H, Freeberg M,  Taylor J,  Nekrutenko A   [*bioRxiv 2018*](https://www.biorxiv.org/content/10.1101/347625v1)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-doi='10.1101/347625'></div> 
1.  **The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update.**  Afgan E, Baker D, Batut B, van den Beek M, Bouvier D, Cech M, Chilton J, Clements D, Coraor N, GrÃ¼ning BA, Guerler A, Hillman-Jackson J, Hiltemann S, Jalili V, Rasche H, Soranzo N, Goecks J, Taylor J, Nekrutenko A, Blankenberg D.   [*Nucleic Acids Res.  2018*](http://www.ncbi.nlm.nih.gov/pubmed/29790989)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='29790989'></div> 
1.  **Child Weight Gain Trajectories Linked To Oral Microbiota Composition.**  Craig SJC, Blankenberg D, Parodi ACL, Paul IM, Birch LL, Savage JS, Marini ME, Stokes JL, Nekrutenko A, Reimherr M, Chiaromonte F, Makova KD.   [*Sci Rep.  2018*](http://www.ncbi.nlm.nih.gov/pubmed/30232389)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='30232389'></div> 
1.  **Predicting runtimes of bioinformatics tools based on historical data: Five years of Galaxy usage.**  Tyryshkina A, Coraor N, Nekrutenko A.   [*Bioinformatics.  2019*](http://www.ncbi.nlm.nih.gov/pubmed/30698642)   <div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='30698642'></div> 
1.  <a id="now"></a>**A high resolution view of adaptive event dynamics in a plasmid**  Mei H, Arbeithuber B, Cremona MA, DeGiorgio M, Nekrutenko A.   [*Genome Biol Evol.  2019*](https://www.ncbi.nlm.nih.gov/pubmed/31539047)<div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='31539047'></div> 
1.  <a id="now"></a>**Bottleneck and selection in the germline and maternal age influence transmission of mitochondrial DNA in human pedigrees.**  Zaidi AA, Wilton PR, Su MS, Paul IM, Arbeithuber B, Anthony K, Nekrutenko A, Nielsen R, Makova KD.   [*PNAS  2019*](https://www.ncbi.nlm.nih.gov/pubmed/31757848)<div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='31757848'></div> 
1.  <a id="now"></a>**HyPhy 2.5-A Customizable Platform for Evolutionary Hypothesis Testing Using Phylogenies**  Kosakovsky Pond SL, Poon AFY, Velazquez R, Weaver S, Hepler NL, Murrell B, Shank SD, Magalis BR, Bouvier D, Nekrutenko A, Wisotsky S, Spielman SJ, Frost SDW, Muse SV.  [*Mol. Biol. Evol.  2019*](https://www.ncbi.nlm.nih.gov/pubmed/31504749)<div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='31504749'></div> 
1.  <a id="now"></a>**Family reunion via error correction: an efficient analysis of duplex sequencing data.**  Stoler N, Arbeithuber B, Povysil G, Heinzl M, Salazar R, Makova KD, Tiemann-
Boege I, Nekrutenko A. [*Genome Biol. 2020*](https://www.ncbi.nlm.nih.gov/pubmed/32131723)<div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='32131723'></div> 
1.  <a id="now"></a>**In memory of James Taylor: the birth of Galaxy.**  Nekrutenko A, Schatz MC. [*Genome Biol. 2020*](https://www.ncbi.nlm.nih.gov/pubmed/32354350)<div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='32354350'></div> 
1.  <a id="now"></a>**The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2020 update.**  Jalili V, Afgan E, Gu Q, Clements D, Blankenberg D, Goecks J, Taylor J,
Nekrutenko A. [*Nucleic Acids Res. 2020*](https://www.ncbi.nlm.nih.gov/pubmed/32479607)<div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='32479607'></div>
1.  <a id="now"></a>**No more business as usual: Agile and effective responses to emerging pathogen threats require open data and open analytics.**  Baker D, van den Beek M, Blankenberg D, Bouvier D, Chilton J, Coraor N, Coppens F, Eguinoa I, Gladman S, GrÃ¼ning B, Keener N, LariviÃ¨re D, Lonie A, Kosakovsky Pond S, Maier W, Nekrutenko A, Taylor J, Weaver S. [*PLoS Pathog. 2020*](https://www.ncbi.nlm.nih.gov/pubmed/32790776)<div class='altmetric-embed' data-badge-type='bar' data-badge-popover='right' data-pmid='32790776'></div> 
