---
title: "Architecture of TraDIS and TnSeq reads"
date: 2019-10-11
tags: [snp, VCF, variant calling, mutect,nvc]
author: Anton
show_summary: false 
draft: false
---

TraDIS and TnSeq are different flavors of transposon insertion techniques aimed at identification of essential genes and perturbing gene expression in bacterial system. Here I try to understand the architecture of Illumina reads generated by each technique.

## TraDIS

TraDIS stands for TRAnsposon-Directed Insertion-site Sequencing. Initially developed by [Langridge et al. 2009](https://genome.cshlp.org/content/19/12/2308.full) and applied to *Salmonella enterica*. It was subsequently used in a number of systems including *Caulobacter crescentus* ([Christen et al. 2011](https://www.embopress.org/doi/full/10.1038/msb.2011.58)), *Escherichia coli* ST131 ([Phan et al. 2013](https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003834)), and others.

The paper by [Goodall et al. (2018)](https://mbio.asm.org/content/9/1/e02096-17/) this blog entry is based on used TraDIS to identify essential genes in *E. coli* BW25113. It is the same strain that was used to construct the [Keio collection](https://cgsc.biology.yale.edu/KeioList.php) of deletion mutants. 

### TraDIS schematics

The basic idea of TraDIS is using a shorter derivative of Tn5 transposon (see [Reznikoff 2008](https://www.annualreviews.org/doi/abs/10.1146/annurev.genet.42.110807.091656)), which is non-specific towards the sequence of an insertion site (can potentially insert anywhere).

The original protocol by [Langridge et al. 2009](https://genome.cshlp.org/content/19/12/2308.full) consists of the following steps:

 1. Prepare transposomes by amplifying the transposon sequence, phosphorylating it, and incubating with Tn5 transposase;
 2. Electroporating bacterial cells with trasposomes prepared at the previous step;
 3. Growing transformats and isolating their genomic DNA;
 4. Sonicating genomic DNA to an average size of 300 bp;
 5. Ligating Illumina adapters to sonication fragments;
 6. Running ligating products on gel and isolating fragments in 250-350 bp range.
 7. Amplifying fragments with a transposon-specific forward primer and Illumina-specific reverse primer
 8. These amplification products are them sequenced using a transposon-specific primer close to the insertion boundary.

 Because the size of sonicated fragments is larger than the inserted transposon such library captures adjacent pieces of genomics DNA, that can be mapped back to genome to identify insertion sites.

 Christen et al. 2011](https://www.embopress.org/doi/full/10.1038/msb.2011.58) modified this approach by adding a promoter that can affect the expression of genes adjacent to the insertion site. They also provide a clear illustration of their method in [supplementary materials](https://www.embopress.org/action/downloadSupplement?doi=10.1038%2Fmsb.2011.58&file=msb201158-sup-0001.pdf): 

 <div class="card">
  <img src="/lab_site/images/christen2011.png" width="632px" alt="Fig.2">

  <div class="card-body">
    <p class="card-text"><b>Fig. S1 |</b> Genomic high-resolution transposon scanning. (A). A feature map of the engineered pXMCS2::Tn5Pxyl transposon delivery plasmid is shown. (B). The nested semi-arbitrary PCR strategy used for simultaneous amplification of transposon junctions is shown. Terminal adapters (blue and orange) compatible for Illumina paired-end sequencing are introduced by the arbitrary PCR primers or via linker-sequence adjacent to the transposon O-end. The second-round PCR products are sequenced from both ends using standard paired-end sequencing primers. (C). Base-call frequencies are shown for a representative paired-end sequencing experiment. The 19 bp long transposon O-end as well as the defined penta-nucleotide of the arbitrary primer produce discrete base calls. From <a href="hhttps://www.embopress.org/action/downloadSupplement?doi=10.1038%2Fmsb.2011.58&file=msb201158-sup-0001.pdf">Christen et al. 2011</a></p>
  </div>
</div>

### TraDIS reads from [Goodall et al. (2018)](https://mbio.asm.org/content/9/1/e02096-17/)

As one can see from the schematics the reads obtained from this protocol are complex with fragments of transposon backbone, indexes etc. in addition to the genomic DNA. These need to be identified and trimmed. [Goodall et al. (2018)](https://mbio.asm.org/content/9/1/e02096-17/) performed trimming and submitted trimmed reads to SRA as [PRJEB24436](https://www.ncbi.nlm.nih.gov/sra/?term=PRJEB24436). 

## TnSeq

The protocol involves creation of transposon insertion libraries (based on Himar1 transposase targeting `TA` sites) using constructs containing promoters of various strengths (as well as one promterless construct). Insertions cause the following effects:

| Effects of instertions on gene expresion |
|--------------|
|![](https://media.nature.com/m685/nature-assets/nchembio/journal/v7/n10/images/nchembio.643-F1.jpg)|
| From Wang et al. [2011](https://www.nature.com/articles/nchembio.643) |

There are six different constructs that include four constructs with a single promoter (*P*<sub>erm</sub>, *P*<sub>pen</sub>, *P*<sub>cap</sub>, and *P*<sub>tuf</sub>), a construct with two promoters (**Dual** containing *P*<sub>erm</sub> and *P*<sub>pen</sub>), and a promoterless construct **Blunt**. The constructs are shown in pane **D** of the figure below below (this figure is from Santiago et al. [2015](https://doi.org/10.1186/s12864-015-1361-3)):

|     |
|--------|
|![](https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2Fs12864-015-1361-3/MediaObjects/12864_2015_1361_Fig4_HTML.gif)|
|(From Santiago et al. [2015](https://doi.org/10.1186/s12864-015-1361-3)). Protocol for the preparation of a high quality transposon DNA library for NGS. (A) (1) Genomic DNA is isolated and digested with NotI. High molecular weight DNA is selectively precipitated using an 8%PEG + NaCl solution, and transposon-plasmid junctions (106 bp) are removed in the supernatant. A biotinylated dsDNA adapter with NotI overhang is ligated (2) before digestion with MmeI, which cuts non-specifically 20 bp from its recognition site within ITR2 into the genome to liberate biotinylated-transposon-genome junctions as short DNA fragments (114 bp). (3). Biotinylated fragments are bound to streptavidin beads (4), and an Illumina sequencing primer adapter containing an indexing barcode and MmeI compatible ends is ligated (5). Primers annealing to the P7 site and the Illumina sequencing primer adapter sequence (with a P5 site overhang) are used to PCR amplify the transposon-genome junctions (6), agarose gel purified, and submitted for Illumina sequencing (7). NGS reads capture both the 16-bp of flanking genomic DNA as well as the transposon donor specific barcode located between the P7 and ITR2. (B) Fragments arising from transposon-plasmid junctions are removed by size selective PEG-NaCl precipitation, while the remaining fragments lack both P7 annealing sites and MmeI sites for ligation of the Illumina sequencing primer adapter. These fragments are therefore not amplified in step (6) of 4A. (C) By performing the size-selective precipitation on a 1 kb DNA ladder, we show that small 300 bp fragments of DNA are retained in the solution (SN), while larger DNA is precipitated (P). (D) Six transposon donor constructs were multiplexed and designed to attenuate expression of genes proximal to the insertion site according to the regulatory elements located at the ends of the transposon backbone. Each donor can be identified from NGS reads by the unique 3 bp barcode.|

### The construct

After digestion, adaptor ligation and additional steps shown in the above figure the following reads are generated:

| Read structure |
|----------------|
|![](/lab_site/images/tn_construct.png)|
|The key components of the read are the Sample Barcode (`Tn seq index 1`), constrcut barcode (`3 bp phage promoiter barcode`) and the `gDNA insert`. The goal of data processing is to parse the two levels of barcodes and trip the read down to the dDNA insert. The insert is then mapped against the genome and the insertion data is analyzed.|

### The reads

In this particular experiment the reads derived from this construct are ~100 - 150 bp long and look like this:

```
1         2              3  4                                               5   6
AGTCATGCA ACACCTAATAAAA  TA ACAGGTTGGATGATAAGTCCCCGGTCTATATTGAGAGTAACTACATT ATC CGTTATGGCACGCGTGGTCTTCGTAT (100 bp)

ACGTANTGT AATTACCGGATATC TA ACAGGTTGGATGATAAGTCCCCGGTCTATATTGAGAGTAACTACATT AAT CGTTATGGCACGCGTGGTCTATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGGGGGGGGGGGGGGG (150 bp)
```

where:

 1. Sample index (BARCODE 1)
 2. Genomic DNA (this is what we will map against the genome)
 3. `TA` instertion site
 4. Transposone backbone that we need to cut out
 5. Promoter-specific barcode (BARCODE 2)
 6. 3'-tail which needs to be trimmed
 
So we need to classify the reads into BARCODE1 &times; BARCODE2 categories and remove all unnecessary stuff.